container.go

```diff
@@ -750,3 +774,45 @@ func (c *Container) detachDevices() error {
 
 	return nil
 }
+
+func (c *Container) addResources() error {
+	//TODO add support for memory, Issue: https://github.com/containers/virtcontainers/issues/578
+	if c.config != nil {
+		cType := ContainerType(c.config.Annotations[vcAnnotations.ContainerTypeKey])
+		if cType == PodSandbox {
+			virtLog.Debugf("Cannot hot add vCPUs to sandbox containers")
+			return nil
+		}
+
+		vCPUs := c.config.Resources.VCPUs
+		if vCPUs != 0 {
+			virtLog.Debugf("hot adding %d vCPUs", vCPUs)
+			if err := c.pod.hypervisor.hotplugAddDevice(uint32(vCPUs), cpuDev); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func (c *Container) removeResources() error {
+	//TODO add support for memory, Issue: https://github.com/containers/virtcontainers/issues/578
+	if c.config != nil {
+		cType := ContainerType(c.config.Annotations[vcAnnotations.ContainerTypeKey])
+		if cType == PodSandbox {
+			virtLog.Debugf("Cannot hot remove vCPUs from sandbox containers")
+			return nil
+		}
```

Comment:
Same comment than above, this is not needed, and more than that, this is preventing the Docker case to work as expected.

Reply:
currently in docker the CPUs are not hot plugged

Reply:
The way I see it, you get the VM started with default CPUs for any case (unless the configuration.toml is saying something else), and every container that we start will be defined with cpu quotas and period, which will help us determine if we have to add some CPUs.